src/components/PopupWithForm.js

```diff
@@ -18,15 +20,27 @@ class PopupWithForm extends Popup {
     return this._formValues;
   }
 
+  setLoadingState(isLoading) {
+    if (this._submitButton) {
+      this._submitButton.textContent = isLoading
+        ? this._defaultButtonText
+        : "Saving...";
+    }
+  }
+
   setEventListeners() {
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(evt, this._getInputValues());
-      this._popupForm.reset();
+      // this._popupForm.reset();
     });
 
     super.setEventListeners();
   }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
 }
 
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```